wit/http2-handler.wit

```diff
@@ -0,0 +1,57 @@
+/// This interface defines a handler of incoming HTTP/2 Requests. It should
+/// be exported by components which can respond to HTTP/2 Requests.
+///
+/// Note: This interface specifically enforces use of HTTP/2, and must not be
+/// provided in environments that aren't able to provide an HTTP/2 connection
+/// or otherwise implement the full semantics of an HTTP/2 connection.
+@since(version = 0.2.2)
+interface http2-incoming-handler {
+  @since(version = 0.2.2)
+  use types.{incoming-request, response-outparam};
+
+  /// This function is invoked with an incoming HTTP Request, and a resource
+  /// `response-outparam` which provides the capability to reply with an HTTP
+  /// Response. The response is sent by calling the `response-outparam.set`
+  /// method, which allows execution to continue after the response has been
+  /// sent. This enables both streaming to the response body, and performing other
+  /// work.
+  ///
+  /// The implementor of this function must write a response to the
+  /// `response-outparam` before returning, or else the caller will respond
+  /// with an error on its behalf.
+  @since(version = 0.2.2)
```

Comment:
This PR implements an interface which has not yet gone through a WASI SG vote, and until it has it can't be included as part of a WASI release. Can you please change the `@since` gates to `@unstable` gates instead?:
```suggestion
  @unstable(feature = "http-http2")
```

Once this PR lands, please also file a PR on the [WASI proposal tracker](https://github.com/WebAssembly/WASI/blob/main/Proposals.md#phase-2---proposed-spec-text-available-cg--wg) so we can track this extension as part of the WASI phrase process. Full details on the process for adding interfaces to existing WASI proposals can be found [here](https://github.com/WebAssembly/WASI/blob/main/Contributing.md#filing-changes-to-existing-phase-3-proposals). Thanks!

Reply:
Oh of course yes: I'll make sure to change this before this lands—if it ever does. Thank you!